core/src/stream_id.rs

```diff
@@ -101,12 +101,13 @@ impl StreamId {
     }
 
     /// Convert the stream id to a vector
-    pub fn to_vec(&self) -> anyhow::Result<Vec<u8>> {
+    pub fn to_vec(&self) -> Vec<u8> {
         // Use self.len() here when we have cid@0.10
         let buf = Vec::new();
         let mut writer = std::io::BufWriter::new(buf);
-        self.write(&mut writer)?;
-        Ok(writer.into_inner()?)
+        // safe to unwrap because self.write should never fail.
```

Comment:
@nathanielc let me know if this is what you had in mind or if there's more I should do to handle this error.

Reply:
LGTM